markdown/src/plugin.c

```diff
@@ -162,9 +162,10 @@ void plugin_help(void)
  * editor's text contents change and not on other editor events.
  */
 static void
-update_markdown_viewer(MarkdownViewer *viewer)
+update_markdown_viewer(MarkdownViewer *viewer, GeanyDocument *doc)
 {
-  GeanyDocument *doc = document_get_current();
+  if (!DOC_VALID(doc))
```

Comment:
Wouldn't simply `doc == NULL` work here?  Do you really get non-NULL doc pointers that have `doc->valid == FALSE`?